Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -69,6 +69,16 @@ open class BasicFormat: SyntaxRewriter {
     super.init(viewMode: viewMode)
   }
 
+  /// Clears all stateful data from this `BasicFormat`.
+  ///
+  /// This needs to be called between multiple `rewrite` calls to a syntax tree.
+  public func reset() {
+    self.indentationStack = [self.indentationStack.first!]
+    self.anchorPoints = [:]
+    self.previousToken = nil
+    self.stringLiteralNestingLevel = 0
```

Comment:
NIT-PICK ALERT!

From what I can see we are not using `self` in other places in this class. 
```suggestion
    indentationStack = [indentationStack.first!]
    anchorPoints = [:]
    previousToken = nil
    stringLiteralNestingLevel = 0
```

Reply:
Oh, if only we were consistent about using `self` or not using `self`. But good catch, I removed it.

Reply:
Doesn't swift-format has a rule for that? 🤔

Reply:
No, because knowing whether `self` requires semantic knowledge. For example you can’t omit the `self` if there’s also a local variable with the same name.